tests/jballerina-unit-test/src/test/resources/test-src/expressions/mappingconstructor/spread_op_field_semantic_analysis_negative.bal

```diff
@@ -71,6 +71,52 @@ function testFieldWithNeverType() {
     Foo _ = {...rec5, ...rec6};
 }
 
+type Address record {
+    string street;
+};
+
+type Street record {|
+    string street;
+|};
+
+function testSpreadOfOpenRecordToCreateClosedRecord() {
+    Address address1 = {street: "Main Street"};
+    record {|string street;|} _ = {...address1};
+
+    Address address2 = {street: "Pine Street", "number": 40};
+    record {|string street;|} _ = {...address2};
+
+    Address address3 = {street: "Maple street" };
+    Street _ = {...address3};
+
+    record {string street;} address4 = {street: "Jump Street"};
+    Street _ = {...address4};
+
+    record {string street;} address5 = {street: "Willow Street", "number": 60};
+    Street _ = {...address5};
+
+    record {string s;} foo1 = {s: "S"};
+    record {int i;} foo2 = {i: 2};
+    Foo _ = {...foo1, ...foo2};
```

Comment:
Let's add a test with multiple compatible and incompatible spread fields when the record is open with an explicit rest descriptor.